python/astro_metadata_translator/translators/decam.py

```diff
@@ -248,3 +251,44 @@ def to_detector_name(self):
         # Docstring will be inherited. Property defined in properties.py
         name = self.to_detector_unique_name()
         return name[1:]
+
+    @classmethod
+    def fix_header(cls, header, obsid, filename=None):
+        """Fix DECam headers.
+
+        Parameters
+        ----------
+        header : `dict`
+            The header to update.  Updates are in place.
+        obsid : `str`
+            Unique observation identifier associated with this header.
+            Will always be provided.
+        filename : `str`, optional
+            Filename associated with this header. May not be set since headers
+            can be fixed independently of any filename being known.
+
+        Returns
+        -------
+        modified = `bool`
+            Returns `True` if the header was updated.
+
+        Notes
+        -----
+        Fixes the following issues:
+        * If OBSTYPE contains "zero" or "bias",
+          update the FILTER keyword to "solid plate 0.0 0.0".
+
+        Corrections are reported as debug level log messages.
+        """
+        modified = False
+
+        obsid = header.get("OBSID", "unknown")
+        obstype = header.get("OBSTYPE", "unknown")
+
+        if "bias" in obstype.lower() or "zero" in obstype.lower():
```

Comment:
DECam has obstypes that have extra text in them above a simple "bias" and "zero"?

Reply:
omg yes, sometimes it's a "postflat-bias" or a "preflat-bias" or a "BIAS" or a "zero" or some camelcase combination of all the above